File: src/main/java/net/ehicks/bts/beans/Status.java
package net.ehicks.bts.beans;

import net.ehicks.bts.ISelectTagSupport;
import net.ehicks.eoi.EOI;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Arrays;
import java.util.List;

@Entity
@Table(name = "statuses")
public class Status implements Serializable, ISelectTagSupport
{
//    @Version
//    @Column(name = "version")
//    private Long version;

    @Id
    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator="ROLE_SEQ")
    @SequenceGenerator(name="ROLE_SEQ", sequenceName="ROLE_SEQ", allocationSize=1)
    @Column(name = "id", updatable = false, nullable = false, columnDefinition = "bigint not null auto_increment primary key")
    private Long id;

    @Column(name = "name", nullable = false, unique = true)
    private String name = "";

    @Override
    public boolean equals(Object obj)
    {
        if (!(obj instanceof Status)) return false;
        Status that = (Status) obj;
        return this.id.equals(that.getId());
    }

    @Override
    public int hashCode()
    {
        return 17 * 37 * id.intValue();
    }

    public String toString()
    {
        return this.getClass().getSimpleName() + ":" + id.toString();
    }

    @Override
    public String getValue()
    {
        return id.toString();
    }

    @Override
    public String getText()
    {
        return name;
    }

    public static List<Status> getAll()
    {
        return EOI.executeQuery("select * from statuses");
    }

    public static Status getById(Long id)
    {
        return EOI.executeQueryOneResult("select * from statuses where id=?", Arrays.asList(id));
    }

    // -------- Getters / Setters ----------

    public Long getId()
    {
        return id;
    }

    public void setId(Long id)
    {
        this.id = id;
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }
}

File: src/main/java/net/ehicks/bts/beans/Status.kt
package net.ehicks.bts.beans

import net.ehicks.bts.ISelectTagSupport
import net.ehicks.bts.Named
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.io.Serializable
import javax.persistence.*

@Entity
data class Status(
        @Id @GeneratedValue(strategy = GenerationType.AUTO)
        private val id: Long = 0,
        @Column(unique = true)
        private val name: String = ""
) : Serializable, ISelectTagSupport, Named {
    override fun getText(): String {
        return name
    }

    override fun getValue(): String {
        return id.toString()
    }

    override fun getId(): Long { return id}
    override fun getName(): String { return name}
}

@Repository
interface StatusRepository : JpaRepository<Status, Long> {
    fun findByName(name: String): Status
    fun findByIdIn(ids: List<Long>): Set<Status>
}